### PR TITLE
feat: refactor project page header components for v4 [1/n]

### PR DESCRIPTION
**Pull request by johnnyd-eth**
- Moves old header components that are essentially V1 only to `packages/v1/`. e.g.:
   - old social links, old `ProjectHeader`
- Moves truly version agnostic header components in V2V3 to `src/components/`. e.g.:
-
- Makes V4 specific `V4ProjectHeader` and `useV4ProjectHeader` with placeholder data 
- Rename some V2V3 specific components e.g. ProjectHeader -> V2V3ProjectHeader

### CODE REVIEW

File: src/packages/v2v3/components/V2V3Project/ProjectDashboard/components/SuccessPayView/components/SuccessPayCard.tsx [hunk 1]
```diff
@@ -8,7 +8,7 @@ import { useProjectPageQueries } from 'packages/v2v3/components/V2V3Project/Proj
 import { useMemo } from 'react'
 import { twMerge } from 'tailwind-merge'
 import { formatCurrencyAmount } from 'utils/format/formatCurrencyAmount'
-import { ProjectHeaderLogo } from '../../ProjectHeader/components/ProjectHeaderLogo'
+import { ProjectHeaderLogo } from '../../../../../../../../components/Project/ProjectHeader/ProjectHeaderLogo'
```
Comment (aeolianeth) [inline src/packages/v2v3/components/V2V3Project/ProjectDashboard/components/SuccessPayView/components/SuccessPayCard.tsx:11]: can probably clean these up

```suggestion
import { ProjectHeaderLogo } from 'components/Project/ProjectHeader/ProjectHeaderLogo'
```

File: src/packages/v4/contexts/V4ProjectMetadataProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+export default function V4ProjectMetadataProvider({
+  projectId,
+  children,
+}: PropsWithChildren<{
+  projectId: bigint
+}>) {
+  const _projectId = parseInt(projectId.toString())
+
+  const { metadata } = useJBProjectMetadataContext()
```
Comment (aeolianeth) [inline src/packages/v4/contexts/V4ProjectMetadataProvider.tsx:15]: very nice

File: src/packages/v4/views/V4ProjectDashboard/hooks/useV4ProjectHeader.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  // const {
+  //   handle,
+  //   projectOwnerAddress,
+  //   totalVolume,
+  //   trendingVolume,
+  //   paymentsCount,
+  //   createdAt,
+  // } = useContext(V2V3ProjectContext) -> How it looks on v2v3
+
+  const handle = 'placeholderData'
```
Comment (aeolianeth) [inline src/packages/v4/views/V4ProjectDashboard/hooks/useV4ProjectHeader.ts:37]: I'd assume that we're not doing V4 handles for the moment.

File: src/packages/v4/views/V4ProjectDashboard/hooks/useV4ProjectHeader.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  const projectOwnerAddress = 'placeholderData'
+  const totalVolume = BigNumber.from(100)
```
Comment (aeolianeth) [inline src/packages/v4/views/V4ProjectDashboard/hooks/useV4ProjectHeader.ts:39]: NOOOOOO bignumber in V4. Sorry, forgot to mention that. BigInt only https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt

```suggestion
  const totalVolume = 100n
```

File: src/utils/routes.ts [hunk 5]
```diff
@@ -14,6 +14,19 @@ export const v2v3ProjectRoute = ({
...
+  projectId,
+  chainName,
+  handle,
+}: {
+  projectId?: BigNumberish
+  chainName?: string
+  handle?: string | null
+}) => {
+  if (handle) return `/@${handle}`
+  return `/v4/${chainName}/p/${BigNumber.from(projectId).toNumber()}`
```
Comment (aeolianeth) [inline src/utils/routes.ts:27]: no bignumber pls

File: src/utils/routes.ts [hunk 6]
```diff
@@ -14,6 +14,19 @@ export const v2v3ProjectRoute = ({
   return `/v2/p/${BigNumber.from(projectId).toNumber()}`
 }
 
+export const v4ProjectRoute = ({
+  projectId,
+  chainName,
+  handle,
+}: {
+  projectId?: BigNumberish
```
Comment (aeolianeth) [inline src/utils/routes.ts:22]:
```suggestion
  projectId?: bigint
```

**Review by aeolianeth — APPROVED** [review]
LGTM, awesome work, this is great. Left a few bignumber comments but otherwise lfg

File: src/packages/v4/views/V4ProjectDashboard/hooks/useV4ProjectHeader.ts (hunk 4 above)
Comment (johnnyd-eth) [reply src/packages/v4/views/V4ProjectDashboard/hooks/useV4ProjectHeader.ts:39]: Oh ok, got it

File: src/packages/v2v3/components/V2V3Project/ProjectDashboard/components/SuccessPayView/components/SuccessPayCard.tsx (hunk 1 above)
Comment (johnnyd-eth) [reply src/packages/v2v3/components/V2V3Project/ProjectDashboard/components/SuccessPayView/components/SuccessPayCard.tsx:11]: Oh yep missed some, got it.

**Pull request merged**